### PR TITLE
Kotlin plugin

### PR DESCRIPTION
**Pull request by sladyn98**

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import io.jenkins.plugins.analysis.core.model.ReportScanningTool;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Provides a parser and customized messages for PEP8 Python style guide.
+ *
+ * @author Sladyn Nunes
+ */
+
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: No newline here

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.parser.JavacParser;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import io.jenkins.plugins.analysis.core.model.ReportScanningTool;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Provides a parser and customized messages for PEP8 Python style guide.
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Update comment

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public static class Descriptor extends ReportScanningToolDescriptor {
+        /** Creates the descriptor instance. */
+        public Descriptor() {
+            super(ID);
+        }
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return Messages.Warnings_JavaParser_ParserName();
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Here you need to add a new Entry to Messages.properties (in src/main/java/io/jenkins/plugins/analysis/warnings). This entry will create a new Java Method that can be referenced here.

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Is there an icon available that can be included in the warnings-plugin? (Check License).

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Provides a parser and customized messages for PEP8 Python style guide.
+ *
+ * @author Sladyn Nunes
+ */
+
+public class Kotlin extends ReportScanningTool {
+
+    private static final long  serialVersionUID = 00; // To be added
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: You can autogenerate one in IntelliJ

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Provides a parser and customized messages for PEP8 Python style guide.
+ *
+ * @author Sladyn Nunes
+ */
+
+public class Kotlin extends ReportScanningTool {
+
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: No newline here

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 2 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Updated

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 1 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Done :+1

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 5 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Added :+1:

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 6 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Done :+1:

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 4 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: I do not think we can use the android one, so NO

File: src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java (hunk 3 above)
Comment (sladyn98) [reply src/main/java/io/jenkins/plugins/analysis/warnings/Kotlin.java]: Added :+1

**Comment by sladyn98** [comment]
@uhafner Kindly review

**Comment by uhafner** [comment]
The file `kotlin.txt` is also missing in the commits. Please also make sure that [Codacy](https://app.codacy.com/app/uhafner/warnings-ng-plugin/pullRequest?prid=3710719) does not report any new issues.

**Comment by sladyn98** [comment]
> The file `kotlin.txt` is also missing in the commits. Please also make sure that [Codacy](https://app.codacy.com/app/uhafner/warnings-ng-plugin/pullRequest?prid=3710719) does not report any new issues.

@uhafner Should I put the file you sent in as kotlin.txt

**Comment by uhafner** [comment]
Well, you can use this file as kotlin.txt if you like. Your test just breaks because the file is not part of the PR. (As well as your changes in the properties file).

**Comment by sladyn98** [comment]
@uhafner made the changes

File: src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java [hunk 7]
```diff
@@ -737,7 +737,17 @@ public void shouldFindAllJavaIssues() {
         shouldFindIssuesOfTool(2 + 1 + 1 + 2, new Java(), "javac.txt", "gradle.java.log", "ant-javac.txt", "hpi.txt");
     }
 
-    /** Runs the CssLint parser on an output file that contains 51 issues. */
+    /**
+     * Runs the Kotlin parser on several output files that contain 4 issues.
```
Comment (uhafner) [inline src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: contains 1 issue

**Comment by uhafner** [comment]
You can also add a new entry in the ChangeLog file and run the `ToolLister` test `shouldPrintAllRegisteredTools`, then you automatically get an update of the `SUPPORTED_TOOLS` document.

File: src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java (hunk 7 above)
Comment (sladyn98) [reply src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: Where do I upload the logo and do we want only a logo or a log and text with it ?

Comment (sladyn98) [reply src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: Where do I upload the logo and do we want only a logo or a log and text with it ?

**Comment by sladyn98** [comment]
> You can also add a new entry in the ChangeLog file and run the `ToolLister` test `shouldPrintAllRegisteredTools`, then you automatically get an update of the `SUPPORTED_TOOLS` document.

Yeah I will make changes to the https://github.com/jenkinsci/warnings-ng-plugin/blob/master/CHANGELOG.md .
In terms of running the toolLister I did not quite get what you mean could you be a little more specific?

**Comment by uhafner** [comment]
Open the class `ToolLister` in IntelliJ and start the Unit test. Then the supported tools are automatically updated.

File: src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java (hunk 7 above)
Comment (uhafner) [reply src/test/java/io/jenkins/plugins/analysis/warnings/ParsersITest.java]: The logos are in https://github.com/jenkinsci/warnings-ng-plugin/tree/master/src/main/webapp/icons. Please add a line that references the license somehow in the License.txt file. 

You can have a look at the https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/java/io/jenkins/plugins/analysis/warnings/AndroidLint.java to see how icons are used.

If everything is working correctly you will see the result in https://github.com/jenkinsci/warnings-ng-plugin/blob/master/SUPPORTED-FORMATS.md (after running the `ToolLister`).

**Comment by sladyn98** [comment]
Made the changes

**Comment by uhafner** [comment]
Thank you very much for your first time contribution!

🎉

**Comment by sladyn98** [comment]
@uhafner Thanks for the mentorship and advice really appreciate it. Looking forward to contributing more to this plugin.

**Pull request merged**